### PR TITLE
SAMD51 Servo class

### PR DESCRIPTION
**Pull request by GMagician**
- Remove tone.c. Next library release will use a lower interrupt priority so it can be used
- Added servo class, library define is buggy and incompilable (some PR should fix it but are not merged yet)

### CODE REVIEW

**Comment by thinkyhead** [comment]
Applied some patches, so it _does_ compile successfully. 🏆

**Comment by GMagician** [comment]
Tune changes has been merged by Adafruit, I hope they will be added to PIO soon.
I don't know why everyone uses maximum interrupt priority for everything...tune, servo...

**Comment by thinkyhead** [comment]
The `tone` handler should be doing very, very little. It only interrupts at the rate determined by the Hz of the current tone, typically under 5k. So it can afford to have a very high priority and shouldn't cause very much trouble.

**Comment by GMagician** [comment]
Well I posted a PR to lower it to level 5 (like the one used here in Marlin)

**Comment by thinkyhead** [comment]
?!? I can't imagine anything being more important than `tone`. 😱

**Comment by GMagician** [comment]
I'm trying to reuse most of marlin code for servo..Original library code is similar to code in shared...
I'm getting a weird error on prescaler and then it should be ok

**Comment by GMagician** [comment]
I had to change PRESCALER define because it collide with some PRESCALER in samd51 framework.
I had the same issue with WDT, compiler expand the define in a struct field called with the same name.
Also had to add `{}` to surround SYNC macro to prevent compile error
Now should compile ok

**Comment by thinkyhead** [comment]
Whenever we find a HAL symbol that conflicts with core headers (or core headers decide to conflict with us), the best thing to do is rename every instance of that symbol with `HAL_` on the front.

File: Marlin/src/HAL/HAL_AVR/ServoTimers.h [hunk 1]
```diff
@@ -53,8 +53,8 @@
  * --------------------
  */
 
-#define TRIM_DURATION       2   // compensation ticks to trim adjust for digitalWrite delays
-#define PRESCALER           8   // timer prescaler
+#define TRIM_DURATION           2   // compensation ticks to trim adjust for digitalWrite delays
+#define SERVO_TIMER_PRESCALER   8   // timer prescaler
```
Comment (thinkyhead) [inline Marlin/src/HAL/HAL_AVR/ServoTimers.h:57]: This name works too!

**Comment by thinkyhead** [comment]
What's left before this can be merged?

**Comment by GMagician** [comment]
I need to change idx=30 using CMSIS defines and I want to see if it's possible to activate timer2...let you know when ready

**Comment by GMagician** [comment]
Now it should be ok..just missing one thing if more than 12 servos are needed (code was and is not ready for this even if I changed a lot of things to go near such target)
I can't test it because I'm waiting for my shield (hope to have at begin of September)

**Comment by GMagician** [comment]
I got an idea..maybe it can solve multi channel issue, last commit is last...

**Comment by GMagician** [comment]
This may be merged now

**Pull request merged**